tests/serverpod_test_server/test_integration/list_include_test.dart

```diff
@@ -0,0 +1,697 @@
+      stockholm.id!,
+      include: City.include(
+        citizens: Person.includeList(),
+      ),
+    );
+
+    expect(city?.citizens, hasLength(2));
+    expect(city?.citizens?.map((e) => e.id), contains(citizen1.id));
+    expect(city?.citizens?.map((e) => e.id), contains(citizen2.id));
+  });
+
+  test(
+      'Given organizations and people when finding a single organization by id and including the people then only the people attached to the organization is included in the result.',
+      () async {
+    var serverpod = await Organization.db.insertRow(
+      session,
+      Organization(name: 'Serverpod'),
+    );
+    var flutter = await Organization.db.insertRow(
+      session,
+      Organization(name: 'Flutter'),
+    );
+
+    var person1 = await Person.db.insertRow(session, Person(name: 'John Doe'));
+    var person2 = await Person.db.insertRow(session, Person(name: 'Jane Doe'));
+    var person3 = await Person.db.insertRow(session, Person(name: 'Alice'));
+
+    await Organization.db.attach.people(session, serverpod, [person1, person2]);
+    await Organization.db.attach.people(session, flutter, [person3]);
+
+    var organization = await session.dbNext.findById<Organization>(
+      serverpod.id!,
+      include: Organization.include(
+        people: Person.includeList(),
+      ),
+    );
+
+    expect(organization?.people, hasLength(2));
+    expect(organization?.people?.map((e) => e.id), contains(person1.id));
+    expect(organization?.people?.map((e) => e.id), contains(person2.id));
+  });
+
+  test(
+      'Given organizations and people when finding all organizations with the people list included then each organization has their respective people in their lists.',
+      () async {
+    var serverpod = await Organization.db.insertRow(
+      session,
+      Organization(name: 'Serverpod'),
+    );
+    var flutter = await Organization.db.insertRow(
+      session,
+      Organization(name: 'Flutter'),
+    );
+
+    var person1 = await Person.db.insertRow(session, Person(name: 'John Doe'));
+    var person2 = await Person.db.insertRow(session, Person(name: 'Jane Doe'));
+    var person3 = await Person.db.insertRow(session, Person(name: 'Alice'));
+
+    await Organization.db.attach.people(session, serverpod, [person1, person2]);
+    await Organization.db.attach.people(session, flutter, [person3]);
+
+    var organizations = await Organization.db.find(
+      session,
+      orderBy: Organization.t.id,
+      include: Organization.include(
+        people: Person.includeList(),
+      ),
+    );
+
+    expect(organizations, hasLength(2));
+    expect(organizations.first.people, hasLength(2));
+    expect(organizations.first.people?.map((e) => e.id), contains(person1.id));
+    expect(organizations.first.people?.map((e) => e.id), contains(person2.id));
+
+    expect(organizations.last.people, hasLength(1));
+    expect(organizations.last.people?.map((e) => e.id), contains(person3.id));
+  });
+
+  test(
```

Comment:
Something like this?
```
city: {
  citizens: Person[],
  organizations: Organization[],
}
```

But isn't this already easy to see by looking at the include structure in the test? And that is checked by the compiler / test runtime as well!

Reply:
Yes, that could be good!

For a person (and maybe us right now) that is used to working with this, it might be quick to decode the structure from code. But I think a helpful comment would make it even faster. Took me a while to understand the relations as I resorted writing it down with pen and paper 😅